frontend/src/metabase/dashboard/reducers.unit.spec.js

```diff
@@ -227,6 +228,25 @@ describe("dashboard reducers", () => {
         },
       });
     });
+
+    it('should not generate runtime error for missing update data in "metabase/entities/questions/UPDATE" action payload', () => {
```

Comment:
It feels like we're fixing a symptom, not the problem.
I think we should go deeper and find out why the update data is missing and fix that.

When I follow reproduction steps from the issue (cloning this particular dashboard) I get JS errors in the console.
Looks like some data may not be reaching the store because of this. Have you tried investigating these?


#### `fetchDashboardCardMetadata` crashing :arrow_left:

- [screenshot](https://github.com/metabase/metabase/assets/6830683/2080acc3-468d-4bf4-87be-9911363ffb6c)
- notice that `await dispatch(loadMetadataForDashboard(cards));` is never reached - perhaps this is the reason?
- looks related to permissions - we're cloning charts that we don't have writable access to? just guessing


#### Selector crashing

- [screenshot](https://github.com/metabase/metabase/assets/6830683/2d524abf-8571-4789-84ca-831bea92d3b8)
- probably unrelated, but worth noticing